src/controllers/scripting/legacy/controllerscriptinterfacelegacy.cpp

```diff
@@ -1052,3 +1058,288 @@ void ControllerScriptInterfaceLegacy::softStart(int deck, bool activate, double
+    case WellKnownCharsets::IBM285:
+        return convertCharsetInternal(QStringLiteral("IBM285"), value);
+    case WellKnownCharsets::IBM290:
+        return convertCharsetInternal(QStringLiteral("IBM290"), value);
+    case WellKnownCharsets::IBM297:
+        return convertCharsetInternal(QStringLiteral("IBM297"), value);
+    case WellKnownCharsets::IBM420:
+        return convertCharsetInternal(QStringLiteral("IBM420"), value);
+    case WellKnownCharsets::IBM424:
+        return convertCharsetInternal(QStringLiteral("IBM424"), value);
+    case WellKnownCharsets::IBM500:
+        return convertCharsetInternal(QStringLiteral("IBM500"), value);
+    case WellKnownCharsets::IBM_Thai:
+        return convertCharsetInternal(QStringLiteral("IBM-Thai"), value);
+    case WellKnownCharsets::IBM870:
+        return convertCharsetInternal(QStringLiteral("IBM870"), value);
+    case WellKnownCharsets::IBM871:
+        return convertCharsetInternal(QStringLiteral("IBM871"), value);
+    case WellKnownCharsets::IBM918:
+        return convertCharsetInternal(QStringLiteral("IBM918"), value);
+    case WellKnownCharsets::IBM1026:
+        return convertCharsetInternal(QStringLiteral("IBM1026"), value);
+    case WellKnownCharsets::IBM1047:
+        return convertCharsetInternal(QStringLiteral("IBM1047"), value);
+    case WellKnownCharsets::IBM01140:
+        return convertCharsetInternal(QStringLiteral("IBM01140"), value);
+    case WellKnownCharsets::IBM01141:
+        return convertCharsetInternal(QStringLiteral("IBM01141"), value);
+    case WellKnownCharsets::IBM01142:
+        return convertCharsetInternal(QStringLiteral("IBM01142"), value);
+    case WellKnownCharsets::IBM01143:
+        return convertCharsetInternal(QStringLiteral("IBM01143"), value);
+    case WellKnownCharsets::IBM01144:
+        return convertCharsetInternal(QStringLiteral("IBM01144"), value);
+    case WellKnownCharsets::IBM01145:
+        return convertCharsetInternal(QStringLiteral("IBM01145"), value);
+    case WellKnownCharsets::IBM01146:
+        return convertCharsetInternal(QStringLiteral("IBM01146"), value);
+    case WellKnownCharsets::IBM01147:
+        return convertCharsetInternal(QStringLiteral("IBM01147"), value);
+    case WellKnownCharsets::IBM01148:
+        return convertCharsetInternal(QStringLiteral("IBM01148"), value);
+    case WellKnownCharsets::IBM01149:
+        return convertCharsetInternal(QStringLiteral("IBM01149"), value);
+    default:
+        m_pScriptEngineLegacy->logOrThrowError(QStringLiteral("Unknown charset specified"));
+        return QByteArray();
+    }
+}
+
+#if QT_VERSION < QT_VERSION_CHECK(6, 4, 0)
+// Add Byte Order Mark (BOM) to the beginning of the data if it is not already present
+static QByteArray addBom(const QByteArray& data) {
+    static const QByteArray bomBE = QByteArray::fromHex("FEFF");
+    static const QByteArray bomLE = QByteArray::fromHex("FFFE");
```

Comment:
constexpr all the things
```suggestion
    static const QByteArray bomBE = QByteArrayLiteral("\xFE\xFF");
    static const QByteArray bomLE = QByteArrayLiteral("\xFF\xFE");
```